File: Selenium/Activities/Activity5.py
# Set up the Firefox Driver with WebDriverManger
from selenium.webdriver import ActionChains
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.firefox.service import Service as FirefoxService
from webdriver_manager.firefox import GeckoDriverManager

# Set up the Firefox Driver with WebDriverManger
service = FirefoxService(GeckoDriverManager().install())

# Start the Driver
with webdriver.Firefox(service=service) as driver:
    # Set up the actions object
    actions = ActionChains(driver)
    # Open the browser to the URL
    driver.get("https://v1.training-support.net/selenium/input-events")
    print("Title of the page is: ", driver.title)
    # Perform Left click and print the value of the side in the front
    actions.click().pause(1).perform()
    getText = driver.find_element(By.CLASS_NAME, "active").text
    print(getText)
    # Perform Double click to show a random side and print the number
    actions.double_click().pause(1).perform()
    randomSideText = driver.find_element(By.CLASS_NAME, "active").text
    print(randomSideText)
    # Perform Right click and print the value shown on the front of the cube
    actions.context_click().pause(1).perform()
    frontSideText = driver.find_element(By.CLASS_NAME, "active").text
    print(frontSideText)

File: Selenium/Activities/Activity5.java
package Activities;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.interactions.Action;
import org.openqa.selenium.interactions.Actions;

public class Activity5 {
    public static void main(String[] args) throws InterruptedException {
        // Setup the Firefox driver(GeckoDriver)
        System.setProperty(FirefoxDriver.SystemProperty.BROWSER_LOGFILE, "/dev/null");
        WebDriverManager.firefoxdriver().setup();
        // Create a new instance of the Firefox driver
        WebDriver driver = new FirefoxDriver();
        // Initialize Actions object
        Actions builder = new Actions(driver);
        driver.get("https://v1.training-support.net/selenium/input-events");
        Thread.sleep(2000);
        System.out.println("Title of the page: " + driver.getTitle());
        // Perform left click and print value of side in the front
        builder.click().pause(1000).build().perform();
        String getText = driver.findElement(By.className("active")).getText();
        System.out.println(getText);
        // Perform Double click to show a random side and print the number
        builder.doubleClick().pause(1000).build().perform();
        String randomSideText = driver.findElement(By.className("active")).getText();
        System.out.println(randomSideText);
        // Perform Right click and print the value shown on the front of the cube
        builder.contextClick().pause(1000).build().perform();
        String frontSideText = driver.findElement(By.className("active")).getText();
        System.out.println(frontSideText);
        // Close the browser
        driver.quit();
    }
}
